org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/ServerModule.java

```diff
@@ -32,5 +34,6 @@ protected void configure() {
 		bind(IMultiRootWorkspaceConfigFactory.class).to(MultiRootWorkspaceConfigFactory.class);
 		bind(IProjectDescriptionFactory.class).to(DefaultProjectDescriptionFactory.class);
 		bind(IContainer.Manager.class).to(ProjectDescriptionBasedContainerManager.class);
+		bind(IRequestManager.class).to(RequestManager.class);
```

Comment:
Smol refinement:
Please bind IRequestManager to the new AbstractRequestManager and that one to RequestManager.

Reply:
It avoids an unfortunately common pitfall. Instead of injecting the interface in a hierarchy Class -> AbstractClass -> Interface, people sometimes accidentally inject the base class.

That being said, it would lead to a runtime error since the base class is abstract in this case. So I'll merge as is.